### PR TITLE
Fix Protosynthesis and Quark Drive boosts 

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Currently both Atk/Sp.Atk and Def/Sp.Def are boosted if either of them is the highest stat.
This should fix it.

## Images
Protosynthesis Nidoqueen.
Before:
pokeemerald-15
After:
pokeemerald-14

## **Discord contact info**
rainonline

### CODE REVIEW

File: src/battle_util.c [hunk 1]
```diff
@@ -8923,18 +8925,19 @@ static u32 CalcMoveBasePowerAfterModifiers(u16 move, u8 battlerAtk, u8 battlerDe
...
     case ABILITY_QUARK_DRIVE:
         {
-            u8 defHighestStat = GetHighestStatId(battlerDef);
-            if (gFieldStatuses & STATUS_FIELD_ELECTRIC_TERRAIN && (defHighestStat == STAT_DEF || defHighestStat == STAT_SPDEF))
+            u8 atkHighestStat = GetHighestStatId(battlerDef);
+            if (gFieldStatuses & STATUS_FIELD_ELECTRIC_TERRAIN
+            && ((IS_MOVE_PHYSICAL(move) && atkHighestStat == STAT_ATK) || (IS_MOVE_SPECIAL(move) && atkHighestStat == STAT_SPATK)))
                 MulModifier(&modifier, UQ_4_12(0.7));
         }
-        break;
```
Comment (LOuroboros) [inline src/battle_util.c:8937]: Please re-add that `break;`.

Comment (AlexOn1ine) [reply src/battle_util.c:8937]: Oh...

Fixed it.

**Comment by DizzyEggg** [comment]
I'd like to see a test added for this 👀

**Comment by AlexOn1ine** [comment]
Added tests and fixed a bug I introduced. The Tests fail without this PR.

Also might be a good Idea to replace the mons once gen9 species are merged.

**Review by DizzyEggg — APPROVED** [review]

**Pull request merged**